appinfo/Migrations/Version20180123143343.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * @author Semih Serhat Karakaya <karakayasemi@itu.edu.tr>
+ * @copyright Copyright (c) 2018, ownCloud GmbH.
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ */
+namespace OCA\password_policy\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use OCP\Migration\ISchemaMigration;
+
+class Version20180123143343 implements ISchemaMigration {
+
+    /** @var  string */
+    private $prefix;
+
+	public function changeSchema(Schema $schema, array $options) {
+        $this->prefix = $options['tablePrefix'];
+        if (!$schema->hasTable("{$this->prefix}user_password_history")) {
+            $table = $schema->createTable("{$this->prefix}user_password_history");
+            $table->addColumn('uid', 'string', [
+                'notnull' => true,
+                'length' => 64,
+            ]);
+            $table->addColumn('password', 'text', [
```

Comment:
stored as clear text? if so this is a no go - please store it hashed

Reply:
no need to worry, I am storing as hashed